packages/main/src/plugin/provider-registry.ts

```diff
@@ -1,5 +1,5 @@
 /**********************************************************************
```

Comment:
remark (non-blocking): the `provider-registry.ts` is 1400 lines long, this will become harder and harder to maintain, and more and more complex to test.

The `provider-registry.spec` is 1700 lines long.